src/utils.js

```diff
@@ -1,3 +1,46 @@
+const NON_STRING_PLACEHOLDER = "__REACT_EMOJI_RENDER_PLACEHOLDER__";
+
 export function escapeStringToBeUsedInRegExp(string) {
   return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"); // $& means the whole matched string
 }
+
+export function stripNonStringElements(children) {
+  if (Array.isArray(children)) {
+    const stripped = children
+      .map(x => (typeof x === "string" ? x : NON_STRING_PLACEHOLDER))
+      .join("");
+    return [stripped, children.filter(x => typeof x !== "string")];
+  } else if (typeof children === "string") {
+    return [children, []];
+  }
```

Comment:
I fear we miss out on some values here being too restrictive about only 2 possibilities for the `children` (and I believe React `children` could be other types than `Array` or `string`).

I believe we should be doing:

```js
if (Array.isArray(children)) {
    ....
}

return [children, []];
```

Reply:
I did some more tests and I think `children` is always an array, the contents of the array can either be a string or an object, sometimes a null. The only case where `children` is not an array is when it's undefined (e.g no children were passed).
So I believe we should remove the condition outside of the `stripNonStringElements` function and instead check if children is an array before calling the function. If `children` is not an array then `text` should stay as it is.
Maybe throw an error if both `text` and `children` are undefined.

Reply:
All right, I think best thing we can do is write test cases for each of those cases and see how it behaves 😁